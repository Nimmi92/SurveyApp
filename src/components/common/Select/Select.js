import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledSelect = styled.div`

 label {
 	margin: 0 5px 0 0;
 }

 select {
  height: 30px;
  padding: 5px 10px;
  background: #e8e8e8;
  border-radius: 3px;
  border: 1px solid #ccc;
 }

`
const Select = (props) => {
 const {data} = props;
  return (
   <StyledSelect>
  	 {data.label && <label>{data.label}</label>}
      <select onChange={data.onChange}>
      {data.options.map((o,i) => {
    	return <option value={o} key={i}>{o}</option>
      })}
     </select>
   </StyledSelect>
  );
};


const { string, func, array } = PropTypes;

Select.propTypes = {
	options: array.isRequired,
	onChange: func,
	label: string,
}

export default Select;