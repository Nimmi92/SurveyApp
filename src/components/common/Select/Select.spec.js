import React from 'react';
import { shallow } from 'enzyme';
import Select from './Select';
import toJson from "enzyme-to-json";

describe('<Select />', () => {
  it('should be an select box', () => {
    const props = {
      label: 'Filter by',
      options: ['select','one','two'],
      onChange: jest.fn()
    };

    const wrapper = shallow(<Select {...props} />);
    const selectBox = wrapper.find('select');
    expect(selectBox.length).toEqual(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

})