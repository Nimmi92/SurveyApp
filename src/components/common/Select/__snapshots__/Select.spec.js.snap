// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should be an select box 1`] = `
<ForwardRef>
  <label>
    Filter by
  </label>
  <select
    onChange={[MockFunction]}
  >
    <option
      key="0"
      value="select"
    >
      select
    </option>
    <option
      key="1"
      value="one"
    >
      one
    </option>
    <option
      key="2"
      value="two"
    >
      two
    </option>
  </select>
</ForwardRef>
`;
